prov/util/src/util_mr_map.c

```diff
@@ -108,22 +108,32 @@ int ofi_mr_map_verify(struct ofi_mr_map *map, uintptr_t *io_addr,
 	void *addr;
 
 	node = ofi_rbmap_find(map->rbtree, &key);
-	if (!node)
-		return -FI_EINVAL;
+	if (!node) {
+                FI_DBG(map->prov, FI_LOG_MR, "verify_addr: unknown key: %" PRIu64 "\n",
+                  key);
+	        return -FI_EINVAL;
+        }
 
 	attr = node->data;
 	assert(attr);
 
 	if ((access & attr->access) != access) {
-		FI_DBG(map->prov, FI_LOG_MR, "verify_addr: invalid access\n");
+                FI_DBG(map->prov, FI_LOG_MR, "verify_addr: invalid access: %s and %s\n",
+                  fi_tostr(&attr->access, FI_TYPE_MR_MODE),
+                  fi_tostr(&access, FI_TYPE_MR_MODE));
 		return -FI_EACCES;
 	}
 
 	addr = (void *) (*io_addr + (uintptr_t) attr->offset);
 
-	if ((addr < attr->mr_iov[0].iov_base) ||
-	    (((char *) addr + len) > ((char *) attr->mr_iov[0].iov_base +
-			    	      attr->mr_iov[0].iov_len))) {
+	if (addr < attr->mr_iov[0].iov_base) {
+                FI_DBG(map->prov, FI_LOG_MR, "verify_addr: access hit before mr range\n");
```

Comment:
I'd merge the checks back together and modify the text.  Something similar to "target region (0x%zx - 0x%zx) out of registered range (0x%zx - 0x%zx)", addr, addr + len, iov_base, iov_base + iov_len)